test/src/org/labkey/test/components/cds/ActiveFilterDialog.java

```diff
@@ -82,18 +83,31 @@ public void saveGroup()
 
     public ActiveFilterDialog saveExpectingError(String errorMsg)
     {
+        final int RETRY_LIMIT = 5;
+        boolean worked = false;
+        int count = 0;
+        String errMsg = "";
         getWrapper().shortWait().until(ExpectedConditions.elementToBeClickable(elementCache().saveGroup));
-        elementCache().saveGroup.click();
-        try
+        while (!worked)
         {
-            getWrapper().waitForElement(elementCache().errorMsg);
+            count++;
+            try
+            {
+                elementCache().saveGroup.click();
+                errMsg = elementCache().errorMsg.findElement(_activeFilterDialogEl).getText();
+                worked = true;
+            }
+            catch (NoSuchElementException ex)
```

Comment:
Is this to catch the 'saveGroup' button not being there or the 'errorMsg' not being there? When I look at the code it has replaced I'm not completely sure which one was the issue.

Ideally you wouldn't want to have a retry like this (although they do exist in other test code). If this is working I'm wondering if part of the success comes from the sleep in the catch. Have you tried explicitly moving the mouse to the button, sleeping for the 1/2 second then doing the click? I know the click should automatically move the mouse to the button but it might be worth trying.